src/AzureIoTHub.Portal/Server/Helpers/DeviceHelper.cs

```diff
@@ -204,5 +205,44 @@ public static IReadOnlyCollection<IoTEdgeModule> RetrieveModuleList(Twin twin)
 
             return list.AsReadOnly();
         }
+
+        /// <summary>
+        /// Convert properties with dot Notation to Twin Collection
+        /// </summary>
+        /// <param name="propertiesWithDotNotation">Properties with dot Notation</param>
+        /// <returns>Twin Collection</returns>
+        public static TwinCollection PropertiesWithDotNotationToTwinCollection(Dictionary<string, object> propertiesWithDotNotation)
+        {
+            var root = new Dictionary<string, object>();
+
+            foreach (var propertyWithDotNotation in propertiesWithDotNotation)
+            {
+                var hierarcy = propertyWithDotNotation.Key.Split('.');
+
+                var current = root;
+
+                for (var i = 0; i < hierarcy.Length; i++)
+                {
+                    var key = hierarcy[i];
+
+                    if (i == hierarcy.Length - 1)
+                    {
+                        // Last element
+                        current.Add(key, propertyWithDotNotation.Value);
+                    }
+                    else
+                    {
+                        if (!current.ContainsKey(key))
+                        {
+                            current.Add(key, new Dictionary<string, object>());
+                        }
+
+                        current = (Dictionary<string, object>)current[key];
+                    }
+                }
+            }
+
+            return new TwinCollection(JsonConvert.SerializeObject(root));
+        }
     }
```

Comment:
This method should be tested independently

Reply:
Done